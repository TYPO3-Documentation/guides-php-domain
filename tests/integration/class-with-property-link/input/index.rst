=========================
PHP class with properties
=========================

..  php:class:: JollyElf

    Welcome to the magical workshop of JollyElves, spreading cheer and joy!

    ..  php:property:: $festiveHatColor
        :type: string
        :public:

        The color of the elf's festive hat, radiating holiday spirit. Can
        Improve the :php:property:`JollyElf::$giftCount`.

    ..  php:property:: $giftCount
        :type: int
        :protected:

        The count of gifts wrapped by the JollyElf, a closely guarded secret.

See also :php:attr:`JollyElf::$festiveHatColor`.
