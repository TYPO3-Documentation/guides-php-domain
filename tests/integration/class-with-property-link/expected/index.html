<!-- content start -->
    <div class="section" id="php-class-with-properties">
            <h1>PHP class with properties</h1>
            <dl class="php class">
    <dt class="sig sig-object php" id="jollyelf">
        <em class="property"><span class="pre">class</span> </em>
        <span class="sig-name descname"><span class="pre">JollyElf</span></span> <a class="headerlink" href="#jollyelf" title="Permalink to this definition">¶</a>
    </dt>
    <dd>

    <p>Welcome to the magical workshop of JollyElves, spreading cheer and joy!</p>
<dl class="php property">
    <dt class="sig sig-object php" id="jollyelf-festivehatcolor">
        <span class="sig-name modifier"><span class="pre">public</span></span> <span class="pre">string</span> <span class="sig-name descname"><span class="pre">$festiveHatColor</span></span>
         <a class="headerlink" href="#jollyelf-festivehatcolor" title="Permalink to this definition">¶</a>
    </dt>
    <dd>
    <p>The color of the elf&#039;s festive hat, radiating holiday spirit. Can
Improve the <a href="/index.html#jollyelf-giftcount">JollyElf::$giftCount</a>.</p>
</dd>
</dl><dl class="php property">
    <dt class="sig sig-object php" id="jollyelf-giftcount">
        <span class="sig-name modifier"><span class="pre">protected</span></span> <span class="pre">int</span> <span class="sig-name descname"><span class="pre">$giftCount</span></span>
         <a class="headerlink" href="#jollyelf-giftcount" title="Permalink to this definition">¶</a>
    </dt>
    <dd>
    <p>The count of gifts wrapped by the JollyElf, a closely guarded secret.</p>
</dd>
</dl>
    </dd>
</dl>

    <p>See also <a href="/index.html#jollyelf-festivehatcolor">JollyElf::$festiveHatColor</a>.</p>

    </div>
<!-- content end -->
